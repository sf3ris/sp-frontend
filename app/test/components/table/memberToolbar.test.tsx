import React from 'react';
import { shallow } from 'enzyme';
import MemberToolbarComponent from '../../../src/components/Table/memberToolbar.component';

describe('Member toolbar component', () => {

    const onNewFn = jest.fn();

    it('Should match snapshot', () => {

        const wrapper = shallow(
            <MemberToolbarComponent 
                onNew={onNewFn} />
        );

        expect(wrapper.debug()).toMatchSnapshot();

    })

    it('Should trigger new', () => {

        const wrapper = shallow(
            <MemberToolbarComponent 
                onNew={onNewFn} />
        );

        const button = wrapper.find('#idNewButtonMemberTable');
        button.simulate('click');

        expect(onNewFn).toBeCalled();

    })
  
})
