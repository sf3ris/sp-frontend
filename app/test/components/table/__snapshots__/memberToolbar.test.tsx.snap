// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member toolbar component Should match snapshot 1`] = `
"<WithStyles(ForwardRef(Grid)) container={true}>
  <WithStyles(ForwardRef(Grid)) item={true} xs={12}>
    <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-1\\" variant=\\"contained\\" id=\\"idNewButtonMemberTable\\" onClick={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} color=\\"primary\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
       
      <span style={{...}}>
        Nuovo
      </span>
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-1\\" variant=\\"contained\\" id=\\"idPDFButtonMemberTable\\" onClick={[Function: togglePDFModal]} color=\\"primary\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
       
      <span style={{...}}>
        PDF
      </span>
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-1\\" variant=\\"contained\\" id=\\"idImportButtonMemberTable\\" onClick={[Function: toggleImportModal]} color=\\"primary\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
      <span style={{...}}>
        Import
      </span>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
  <MemberPDFModal onOk={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} isOpen={false} toggle={[Function: togglePDFModal]} />
  <ImportModal onImportMembers={[Function: onImportModal]} isOpen={false} toggle={[Function: toggleImportModal]} />
</WithStyles(ForwardRef(Grid))>"
`;
